config/config_builder/src/swarm_config.rs

```diff
@@ -12,6 +12,39 @@ use failure::prelude::*;
 use nextgen_crypto::{ed25519::*, test_utils::KeyPair};
 use std::path::{Path, PathBuf};
 
+/// Topology indicates the shape of the validator network
+/// Currently does not handle full nodes, the launch_swarm will only use num_nodes value
+#[derive(Debug, Clone)]
+pub struct LibraSwarmTopology {
+    // TODO: make it more general to support various network shapes
+    data: Vec<usize>,
+}
+
+impl LibraSwarmTopology {
+    pub fn create_validator_network(num_validator_nodes: usize) -> Self {
+        Self {
+            data: vec![num_validator_nodes],
+        }
+    }
+
+    pub fn create_uniform_network(num_validator_nodes: usize, num_full_nodes: usize) -> Self {
+        Self {
+            data: vec![num_validator_nodes, num_full_nodes],
+        }
+    }
+
+    pub fn num_validators(&self) -> usize {
+        self.data[0]
```

Comment:
nit: technically better to check for a size here as well

Reply:
will check for emptiness